openedx/core/djangoapps/programs/utils.py

```diff
@@ -463,26 +463,44 @@ def _collect_one_click_purchase_eligibility_data(self):
         if is_learner_eligible_for_one_click_purchase:
             for course in self.data['courses']:
                 add_course_sku = False
+                unpublished_enrollment = False
+                unpublished_course_runs = filter(lambda run: run['status'] == 'unpublished', course['course_runs'])
```

Comment:
Pretty sure this could be a list comprehension (as talked about in yesterdays AMA :))

unpublished_course_runs = [run for run in course['course_runs'] if run['status'] == 'unpublished']

maybe it is or isnt more readable.

Reply:
I prefer it to match what is already there.  I do not lean in one camp or the other.